### PR TITLE
feat: Integrate cdk nag

### PR DESCRIPTION
**Pull request by peermuellerxw**
Also replaced deprecated rules from typescript-eslint, see https://typescript-eslint.io/blog/deprecating-formatting-rules/

### CODE REVIEW

File: cdk/package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
   "devDependencies": {
     "@commitlint/cli": "^19.0.3",
     "@commitlint/config-conventional": "^19.0.3",
+    "@stylistic/eslint-plugin": "^1.7.2",
```
Comment (aperov9) [inline cdk/package.json:17]: Is this a requirement for cdk nag?

File: cdk/src/stacks/backup.ts [hunk 2]
```diff
@@ -6,8 +6,10 @@ import { Construct } from 'constructs';
 export class BackupStack extends NestedStack {
   constructor(scope: Construct, id: string, props: NestedStackProps) {
     super(scope, id, props);
-    new CfnInclude(this, 'SuperwerkerTemplate', {
+    const cfnInclude = new CfnInclude(this, 'SuperwerkerTemplate', {
       templateFile: path.join(__dirname, '..', '..', '..', 'templates', 'backup.yaml'),
     });
+
+    cfnInclude.stack.addMetadata('cfn - lint', { config: { ignore_checks: ['E9007', 'EPolicyWildcardPrincipal', 'E1029'] } });
```
Comment (aperov9) [inline cdk/src/stacks/backup.ts:13]: Just need to remember to delete this one after we have migrated the respective CFN stacks to CDK.

**Review by aperov9 — APPROVED** [review]

File: cdk/package.json (hunk 1 above)
Comment (peermuellerxw) [reply cdk/package.json:17]: No, just did this "driving-by".

File: cdk/src/stacks/backup.ts (hunk 2 above)
Comment (peermuellerxw) [reply cdk/src/stacks/backup.ts:13]: Yes, or just not copy over the metadata to the cdk stack ;) 
I think we never did this previously.

Comment (aperov9) [reply cdk/src/stacks/backup.ts:13]: But at the end of the day we don't really need it, cfn lint is more of a tool when writing plain cloudformation. Cdk does pretty decent stuff and nag on top should be sufficient 👍🏻

Comment (peermuellerxw) [reply cdk/src/stacks/backup.ts:13]: Yes, but at the moment it is still enabled via a github action and we still have have a few cfn files.

**Pull request merged**